src/common/models/users.ts

```diff
@@ -34,15 +38,15 @@ export class User extends Model {
 	get all_field_names(): string[] { return User.ALL_FIELDS; }
 	get param_fields(): string[] { return []; }
 
-	constructor(params: ParseableUser = {}) {
+	constructor(params: ParseableUser = { user_id: 0, username: '', is_admin: false }) {
```

Comment:
username: `logged_out`?

Perhaps export this default as a constant for use in session.ts?

Reply:
That's a good idea